neuron.schema.json

```diff
@@ -0,0 +1,148 @@
+    "properties": {
+        "friendlyName": {
+            "description": "A friendly name for the neuron class",
+            "type": "string"
+        },
+        "summaryDescription": {
+            "description": "A description of the function of the neuron",
+            "type": "string"
+        },
+        "className": {
+            "description": "The programmatic name for the class",
+            "type": "string"
+        },
+        "parameters": {
+            "description": "All the parameter names for the simulation",
+            "type": "array",
+            "items": {
+                "type": "string"
+            }
+        },
+        "variables": {
+            "description": "All the variable names and types for the simulation",
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string"
+                    },
+                    "type": {
+                        "type": "string",
+                        "enum": ["scalar","vector"]
+                    }
+                },
+                "required": ["name", "type"],
+                "additionalProperties": false
+            }
+        },
+        "derivedParameters": {
+            "description": "All the names and derivation functions of secondary parameters in the simulation",
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string"
+                    },
+                    "derivation": {
+                        "type": "object",
+                        "minProperties": 1,
+                        "properties": {
+                            "generic": {"type": "string", "minlength": 1},
+                            "python": {"type": "string", "minlength": 1},
+                            "cpp": {"type": "string", "minlength": 1}
+                        },
+                        "additionalProperties": {"type": "string", "minlength": 1}
+                    }
+                },
+                "required": ["name", "derivation"],
+                "additionalProperties": false
+            }
+        },
+        "additionalParameters": {
+            "description": "The names and types of additional parameters for the simulation",
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string"
+                    },
+                    "type": {
+                        "type": "string",
+                        "enum": ["scalar","vector"]
+                    }
+                },
+                "required": ["name", "type"],
+                "additionalProperties": false
+            }
+        },
+        "additionalVariables": {
```

Comment:
Same question - what additional variables?

Reply:
_additional_input_vars_ from Python - not sure of C++ equivalent

Reply:
So these last two things are called "additional input variables" and "extra global parameters" give or take different naming conventions in both Python and C++ so I think they should be called the same here. The input bit is key as additional input variables are used for adding additional (potentially non-linearly combined) inputs to neurons (don't worry @liamstaras , that was more to jog @tnowotny 's memory than for you :smile)

Reply:
The C++ additional input variable syntax is described [here](http://genn-team.github.io/genn/documentation/4/html/de/ded/sectNeuronModels.html#neuron_additional_input)

Reply:
fine, I'll rename to additionalInputVariables and extraGlobalParameters :wink:

Reply:
sounds good - and yes, @neworderofjamie , I do now recall vaguely what this was about! Thanks for the pointers.